### ISSUE TITLE
Upload error code 209/211/212

### ISSUE THREAD

**Issue by RolandFran** [Portisch/RF-Bridge-EFM8BB1#193]
Hi

I tried to upgrade the RF-firmware on a R2 revision of the Sonoff RF Bridge that I got today
with portisch "RF-Bridge-EFM8BB1-20190220.hex".
After about 5 to 10 seconds the upload failed with "Upload error code 209" and later 211 and also 212

---------------------
Console:
00:00:00.049 CFG: Loaded from flash at FB, Count 35
00:00:00.054 SER: Set to 8N1 19200 bit/s
00:00:00.055 QPC: Count 1
00:00:00.060 SER: Set to 8N1 19200 bit/s
00:00:00.063 Project tasmota Tasmota Version 9.3.1(tasmota)-2_7_4_9(2021-03-09T16:12:28)
...
21:56:17.043 UPL: File portisch506.hex
21:56:18.618 UPP: Multicast (re)joined
21:56:18.619 UPL: Transfer 22702 bytes
--------------------
I tried already other jumper cables, cables other way around and switch on instead off -> same result
Any idea what that could be?

**Comment by ricardo-cruz44** [comment]
What card are you using to flash the EFM8Bb?
I have a Sonoff RF. Bridge Rev 2 that I flashed last week no problem.

**Comment by RolandFran** [comment]
yes, that's the same one I have.
Flashing Tasmota was no problem, but the portisch firmware update failed.

**Comment by ricardo-cruz44** [comment]
Sorry for the confusion, on the RF bridge rev2 for flashing the EFM8BB I used an Arduino uno card to control the C2d and C2ck pins and upload the hex file to the EFM8BB, I have not done the Tasmota flash upgrade yet, still deciphering the  capture data.

**Comment by ElwynnEldoriath** [comment]
> yes, that's the same one I have.
> Flashing Tasmota was no problem, but the portisch firmware update failed.

Have you been able to solve the issue? I am having the same errors (code 211) when trying to upload the Portisch firmware.

**Comment by RolandFran** [comment]
Unfortunately not, spent several hours trying to get the bridge working, till now without any success

**Comment by Mindsuk** [comment]
try to flash Tasmota again and than Portisch again

**Comment by ElwynnEldoriath** [comment]
> try to flash Tasmota again and than Portisch again

Did it multiple times, no change... always getting error 211


EDIT: I have now downgraded to 9.1 (fresh firmware flash) and it says "Failed to erase RF chip". Any help?

**Comment by waynen1967** [comment]
Total newbie here but I followed this youtube video and had no issues installing Tasmota and Portisch

https://youtu.be/cApVlG7Rfhk

**Comment by ElwynnEldoriath** [comment]
Watched it before, no difference...

**Comment by cph25** [comment]
I was running in the same issue and very close to give up. The video by DrZzs shows a successful flashing of the RF chip. There was one difference, he used the Tasmota 8.4.0 and the English version, downloaded the 8.4.0  from https://github.com/arendst/Tasmota/releases/, flashed the bridge via Tasmotizer and it worked.

**Comment by ElwynnEldoriath** [comment]
Still getting a "Failed to init RF chip". Either the device is broken or there is a different hardware rev. circling

**Comment by cph25** [comment]
I got this too multiple times: check if the 2 wires are well connected, play with the switch s2 - I'm no more sure when it worked.

**Comment by ElwynnEldoriath** [comment]
I checked the proper connection with a multimeter, tried both switch position: no change

**Comment by chrisvor** [comment]
Just a short note that I was having the exact same issue with Tasmota 9.4.0. Then I saw the post by cph25 mentioning that DrZzs was using Tasmota 8.4.0..... So I thought to give it a try... Reflashed 8.4.0 -> Firmware upgrade -> selected the portisch 2019 hex file -> boom! Flash successful!
I then updated tasmota to 9.4.0 (minimal first, then full version) and I am now on 9.4.0 with portisch firmware working!

(edit: I initially got the "Failed to init RF chip" too, and I swapped the two jumper wires. Switch was in the OFF position I think, if it doesnt work, switch to ON and try again)

**Comment by ktuulos** [comment]
On my case, this worked fine with Tasmota 9.5.0 and RF-Bridge-EFM8BB1-20190220.hex. RF bridge was powered externally with 3.3 volts and GPIO lines were not cut away. There were just two flashing wires between MCU and EFM8BB1.

**Comment by kiat-huang** [comment]
> On my case, this worked fine with Tasmota 9.5.0 and RF-Bridge-EFM8BB1-20190220.hex. RF bridge was powered externally with 3.3 volts and GPIO lines were not cut away. There were just two flashing wires between MCU and EFM8BB1.

Same experience here, except I had the two wires attached as per the instructions...![sonoff_bridge_rffw_upgrade](https://user-images.githubusercontent.com/17239259/133873789-02f8ddc3-99d8-441e-80eb-dd6feef61258.png)

The webui said the Portisch flash worked but I see no signs of it in the Info section.  Am guessing all is fine and it'll be apparent once I start adding certain devices.

**Comment by Robmak73** [comment]
I had success with same setup. The wire cut is not needed if you power the
chip directly if I remember right.
I did brick my first bridge but put this down to the ft232 ttl adapter.
Managed to tamotize straight away cp2102

**Comment by arbasel** [comment]
I have the Sonoff RF Bridge R2 but v2
![20210925_231502](https://user-images.githubusercontent.com/6464534/134769517-665b6fdf-1f8a-41c6-b2db-d235314c2bf8.jpg)
@Robmak73  could you please assist me with which pins I need to connect to where? From what you say I should not need to cut any wires.

**Comment by jgruiz68** [comment]
@arbasel Did you have any luck with the board? I have the same version.

**Comment by Robmak73** [comment]
Yes I used the tasmotizer and the software did everything for me. 
No need to cut wires on the board. Just make sure you follow the power up procedure and connect the 2 sets of 2 pins together. I did brick a sonoff bridge but found that it was due to the usb ttl converter from ebay was the wrong type. Soon as I got the right usb to ttl adapter I had no problem at all.

**Comment by jgruiz68** [comment]
I used tasmotizer to flash Tasmota as well, but did you also flash the portisch firmware using tasmotizer? The part I am having trouble with is flashing the portisch firmware.
> 
> 
> Yes I used the tasmotizer and the software did everything for me. No need to cut wires on the board. Just make sure you follow the power up procedure and connect the 2 sets of 2 pins together. I did brick a sonoff bridge but found that it was due to the usb ttl converter from ebay was the wrong type. Soon as I got the right usb to ttl adapter I had no problem at all.

**Comment by joncknott** [comment]
> I used tasmotizer to flash Tasmota as well, but did you also flash the portisch firmware using tasmotizer? The part I am having trouble with is flashing the portisch firmware.
> 

@jgruiz68 I'm having the same issue unfortunately. From a [message in the Discord channel about Tasmota Flashing](https://discord.com/channels/479389167382691863/790182241908686870/892472267764805664), it seems that the R2 V2.2 boards have a different RF chip:

> As far as I can tell, the old EFM8BB1 has been replaced by an 8R08A

So it looks like this firmware will not work on the Sonoff RF Bridge R2 V2.2. I'd love to be proven wrong though!

**Comment by jgruiz68** [comment]
@joncknott Thanks for the reference, I will keep an eye on new developments. In the mean time I will look for an older bridge.

**Comment by Robmak73** [comment]
No I did not use tasmotizer as it does not do the Portishead flash. Its relatively simple if you get the power correct. The wifi does not have to be on and the hex file is small. I used node mcu as per wiki information. It is really easy if you follow the process exactly Sent from Samsung Mobile on O2
-------- Original message --------From: joncknott ***@***.***> Date: 06/10/2021  19:01  (GMT+00:00) To: Portisch/RF-Bridge-EFM8BB1 ***@***.***> Cc: Robmak73 ***@***.***>, Mention ***@***.***> Subject: Re: [Portisch/RF-Bridge-EFM8BB1] Upload error code 209/211/212 (#193) 

I used tasmotizer to flash Tasmota as well, but did you also flash the portisch firmware using tasmotizer? The part I am having trouble with is flashing the portisch firmware.

@jgruiz68 I'm having the same issue unfortunately. From a message in the Discord channel about Tasmota Flashing, it seems that the R2 V2.2 boards have a different RF chip:

As far as I can tell, the old EFM8BB1 has been replaced by an 8R08A

So it looks like this firmware will not work on this version of the Sonoff Bridge.

**Comment by Robmak73** [comment]
I don't think that is necessary. Just make sure you power the chip directly with 3.3v when flashing rf433 chip. If you look at the circuit board you can see the pin holes for independent power. The wiki explains it much better than me. But the board only needs the 2 conducting strips cutting if you trying to power the wifi chip and rf433 chip on the same adapter. The wifi chip does not need power to flash the rf433 chip. So don't cut the board. Just move your + and - from the wifi input to the rf433 input. The board is labelled if you look closely. To double check follow the track to show the exact same result by using this method to cutting the board. The rf433 chip is much easier to flash. Infact it was done so quickly I thought that it had not worked. The hex (if I remember correctly) file is relatively small. Sent from Samsung Mobile on O2
-------- Original message --------From: Juan Ruiz ***@***.***> Date: 06/10/2021  19:14  (GMT+00:00) To: Portisch/RF-Bridge-EFM8BB1 ***@***.***> Cc: Robmak73 ***@***.***>, Mention ***@***.***> Subject: Re: [Portisch/RF-Bridge-EFM8BB1] Upload error code 209/211/212 (#193) 
@joncknott Thanks for the reference, I will keep an eye on new developments. In the mean time I will look for an older bridge.

**Comment by tomlyo** [comment]
The older model rf bridge is disappearing from online stores. I've got 2 of the newer r2 v2 models and would love to flash this firmware if anybody has figured it out yet.

**Comment by grf21a** [comment]
Hi,  Hope this may help some people experiencing the 209 error.

I flashed my Black RF Bridge today with Tasmota 11.0.0 and then proceeded to install Portisch following the video available by DrZzs.

During the installation of Portisch firmware I was getting the "Upload error code 209" message appearing and went searching for an answer but could not find anything that would help.

I checked that the filename had the correct file extension as mentioned by DrZzs.  For those that do not know already, it is important that within Win10 File Explorer, click on the View menu, then select the tick box in "File Name Extensions" (see attached graphic). You can also go to a windows command prompt and navigate to the folder that contains the file and run the DIR command to double check.

The file name was not the root cause in my case so I opened up the file in Notepad to double check that I has pasted contents correctly and again everything looked okay initially.  I scrolled to the bottom of the file and noticed that cursor was positioned at the end of the last line, so I decided to hit enter on this line to move the cursor to the next line so the file would end on a blank line (see attached)

When I ran the Firmware Upgrade again everything worked successfully, so it ended up being a simple carriage return/line feed that caused my issue.

Anyway, hope this helps other people out there.  Good luck
![ExplorerFileExtensions](https://user-images.githubusercontent.com/99779985/162569859-3516d1a3-5f68-4904-8b3b-342b8c7223a4.jpg)
![Portisch File End](https://user-images.githubusercontent.com/99779985/162569874-6f8d8bde-57f9-4eaa-9980-dcee69b0ea2e.jpg)
.
